### DISCUSSION TITLE
Failing to access a dictionary on the parent's DataContext

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#9070] [Q&A] [answered]
```xml
<ComboBox Items="{Binding SomeArray}" Width="320">
  <ComboBox.ItemTemplate>
    <DataTemplate>
      <ComboBoxItem>
        <ComboBoxItem.Content>
          <StackPanel Orientation="Horizontal">
            <TextBlock Text="{Binding Number}" />
            <TextBlock Text="-" />
            <TextBlock Text="{Binding $parent[ComboBox].DataContext.SomeDictionary[Number]}" />
          </StackPanel>
        </ComboBoxItem.Content>
      </ComboBoxItem>
    </DataTemplate>
  </ComboBox.ItemTemplate>
</ComboBox>
 ```

```cs
public class ExampleObject
{
    public uint Number { get; set; }
}

public class SomeViewModel : ViewModelBase
{
    public ExampleObject[] SomeArray { get; set; } = 
    { 
        new() {Number = 3}, 
        new() {Number = 4} 
    };
     public Dictionary<uint, string> SomeDictionary { get; set; } = new() 
     {
        [1] = "One",
        [2] = "Two"
     };
}
```

I'm trying to access the content of `SomeDictionary`.
Unfortunately the example above does not work as nothing shows in the third `TextBlock` 

No exceptions occur, it just display the following elements:
```
1 -
2 -
```
instead of displaying
```
1 - One
2 - Two
```

however, I can access `$parent[ComboBox].DataContext.SomeArray` without any issues 

Update: I also figured out that making use of `$parent[ComboBox].DataContext` has an underlying issue which is that an error will occur whenever a user switch to another view model

any ideas?

**Comment by Mrcubix** [comment]
I think i was doing it the wrong way but i still don't think the method i ended up using is really appropriate either:

I Implemented a converter that convert a `Number` into the appropriate value from the dictionary

```cs
    public class NumberToText : IValueConverter
    {
        public static Dictionary<uint, string> SomeDictionary { get; set; } = new();

        public object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
        {
            if (value is uint number)
            {
                if (SomeDictionary.ContainsKey(number))
                    return SomeDictionary[number];
            }

            return null;
        }

        public object? ConvertBack(object? value, Type targetType, object? parameter, CultureInfo culture)
        {
            if (value is string number)
            {
                return SomeDictionary.FirstOrDefault(x => x.Value == number).Key;
            }

            return null;
        }
    }
```

**Comment by maxkatz6** [comment, marked as answer]
`{Binding $parent[ComboBox].DataContext.SomeDictionary[Number]}`

It will look up for an entry with "Number" key.

**Reply by maxkatz6** [reply]
IIRC in all XAML frameworks indexers in bindings accept only constant variables. Instead, you can use MultiBinding with two parameters - dictionary and a key binding, or a specific IValueConverter

**Reply by Mrcubix** [reply]
i ended up using an IValueConverter, the issue with that being, i easily have 50 different "hashes" as they call it in the api i'm using, so it's not gonna really work if i have to make as many converters as there are "hashes".

I tried making MultiBinding Work but all i was getting out of it were exceptions, which mean i probably did something wrong, i'll try again later
